src/apps/content-editor/src/app/views/ItemList/ItemList.js

```diff
@@ -56,708 +63,629 @@ export default connect((state, props) => {
-        filters: {
-          filterTerm: this.state.filterTerm,
-          sortedBy: this.state.sortedBy,
-          status: this.state.status,
-          colType: this.state.colType,
-          reverseSort: this.state.reverseSort
-        }
-      });
+  // HACK: runFilters after waiting for new props
+  useEffect(() => {
+    if (_isMounted.current && shouldRunFilters) {
+      runFilters();
+      setShouldRunFilters(false);
     }
+  }, [shouldRunFilters]);
 
-    componentDidUpdate(prevProps) {
-      if (
-        this.props.model &&
-        (this.props.model.name === "clippings" ||
-          this.props.model.type === "templateset")
-      ) {
-        // First item is always the column row
-        if (this.state.items.length === 2) {
-          // redirect to the single entry in content clippings
-          window.location.hash = `/content/${this.props.modelZUID}/${this.state.items[1].meta.ZUID}`;
-        }
-      }
+  // HACK: updateItemCount after waiting for new props
+  useEffect(() => {
+    if (_isMounted.current && shouldUpdateItemCount) {
+      updateItemCount();
+      setShouldUpdateItemCount(false);
+    }
+  }, [shouldUpdateItemCount]);
+
+  // on items update, update itemCount
+  useEffect(() => {
+    setItemCount(Math.max(0, items.length - 1));
+  }, [items]);
+
+  // set table dimensions, re-check on props change
+  useEffect(() => {
+    if (
+      table.current &&
+      (height !== table.current.clientHeight ||
+        width !== table.current.clientWidth)
+    ) {
+      setHeight(table.current.clientHeight);
+      setWidth(table.current.clientWidth);
+    }
+  }, [props]);
+
+  function updateItemsAndFields() {
+    const newFields = findFields(props.allFields, props.modelZUID);
+    const newItems = findItems(
+      props.allItems,
+      props.modelZUID,
+      props.selectedLang.ID
+    );
+
+    newItems.unshift(newFields);
+    setFields(newFields);
+    setItems(newItems);
+  }
 
-      // Handle changing item list views
-      if (prevProps.modelZUID !== this.props.modelZUID) {
-        this.updateRows();
-        // Clear previous model and load any new model rows we have available
-
-        // Load new model from API
-        this.load(this.props.modelZUID);
-        if (this.props.filters[this.props.modelZUID]) {
-          this.runAllFilters();
-        } else {
-          this.setState({
-            filterTerm: "",
-            sortedBy: null,
-            status: "all",
-            colType: null,
-            reverseSort: false
-          });
+  async function load(modelZUID) {
+    setLoading(true);
+    setLoadingSecondary(true);
+
+    try {
+      const [, itemsRes] = await Promise.all([
+        props.dispatch(fetchFields(modelZUID)),
+        props.dispatch(fetchItems(modelZUID, { limit: PAGE_SIZE, page: 1 }))
+      ]);
+      if (_isMounted.current) {
+        // render 1st page of results
+        // HACK: runFilters after waiting for new props
```

Comment:
Is this trying to communicate that filters will be run at the next render cycle vs that code executing immediately?